api/admin_test.go

```diff
@@ -0,0 +1,292 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/netlify/gotrue/models"
+)
+
+// TestAdminUsersUnauthorized tests API /admin/users route without authentication
+func TestAdminUsersUnauthorized(t *testing.T) {
+	api, err := NewAPIFromConfigFile("config.test.json", "v1")
+	if err != nil {
+		t.Error(err)
+	}
+	defer api.db.Close()
+
+	// Setup request
+	req := httptest.NewRequest("GET", "/admin/users", nil)
+
+	// Setup response recorder
+	w := httptest.NewRecorder()
+	ctx := req.Context()
+
+	api.adminUsers(ctx, w, req)
+
+	resp := w.Result()
+
+	if resp.StatusCode == 200 {
+		t.Log(resp)
+		t.Fail()
+	}
+}
+
+func makeSuperAdmin(req *http.Request, api *API, email string, t *testing.T) (context.Context, *httptest.ResponseRecorder) {
```

Comment:
tiny thing (as in feel free to ignore) but require/assert take the `t *testing.T` as the first param. I tend to try to follow that convention, but def not necessary.